File: flutter_boilerplate_dash/lib/module_shipments_orders_accepted/request/shipment_filter_request.dart
class AcceptedShipmentFilterRequest{
  String? trackNumber;
  String? transportationType;
  String? status;
  bool? isExternalWarehouse;
  int? exportWarehouseID;
  int? importWarehouseID;
  int? clientUserID;

  String? paymentTime;
  String? orderStatus;
  // String? launchCountry;
  String? targetCountry;

  String? dateOne;
  String? dateTow;

  String? containerNumber;
  String? airWaybillNumber;


  bool acceptedUntilCleared;

  AcceptedShipmentFilterRequest({required this.acceptedUntilCleared,
    this.transportationType, this.trackNumber , this.status ,
    this.orderStatus,
    this.isExternalWarehouse , this.exportWarehouseID , this.importWarehouseID , this.airWaybillNumber , this.dateTow , this.dateOne , this.paymentTime , this.targetCountry , this.containerNumber , this.clientUserID});

  Map<String, dynamic> toJson() => {
    'trackNumber':trackNumber ==''?null:trackNumber,
    'status':status ==''?null :status,
    'transportationType':transportationType,
    'isExternalWarehouse':isExternalWarehouse,
    'exportWarehouseID':exportWarehouseID,
    'importWarehouseID':importWarehouseID,
    'orderStatus':orderStatus ?? 'accepted',

    'targetCountry':targetCountry ==''? null:targetCountry,
    // 'launchCountry':launchCountry,
    'paymentTime':paymentTime == ''?null : paymentTime,

    'dateOne':dateOne ==''?null : dateOne,
    'dateTow':dateTow == ''?null :dateTow,
    'containerNumber':containerNumber == ''?null:containerNumber ,
    'airWaybillNumber':airWaybillNumber =='' ? null :airWaybillNumber,
    'clientUserID':clientUserID,
    'acceptedUntilCleared':acceptedUntilCleared
  };
}

File: backend/src/Request/ShipmentFilterRequest.php
<?php

namespace App\Request;

class ShipmentFilterRequest
{
    private $status;

    private $transportationType;

    private $trackNumber;

    private $paymentTime;

    private $launchCountry;

    private $targetCountry;

    private $isExternalWarehouse;

    private $exportWarehouseID;

    private $importWarehouseID;

    private $dateOne;

    private $dateTwo;

    private $containerNumber;

    private $airWaybillNumber;

    private $clientUserID;

    private $measuredUntilCleared;

    private $measuredUntilArrived;

    private $acceptedUntilCleared;

    private $acceptedUntilArrived;

    public function getStatus()
    {
        return $this->status;
    }

    public function getTransportationType()
    {
        return $this->transportationType;
    }

    public function getPaymentTime()
    {
        return $this->paymentTime;
    }

    public function getExportWarehouseID()
    {
        return $this->exportWarehouseID;
    }

    public function getLaunchCountry()
    {
        return $this->launchCountry;
    }

    public function getTargetCountry()
    {
        return $this->targetCountry;
    }

    public function getTrackNumber()
    {
        return $this->trackNumber;
    }

    public function getIsExternalWarehouse()
    {
        return $this->isExternalWarehouse;
    }

    public function getImportWarehouseID()
    {
        return $this->importWarehouseID;
    }

    public function getDateOne()
    {
        return $this->dateOne;
    }

    public function getDateTwo()
    {
        return $this->dateTwo;
    }

    public function getContainerNumber()
    {
        return $this->containerNumber;
    }

    public function getAirWaybillNumber()
    {
        return $this->airWaybillNumber;
    }

    public function getClientUserID()
    {
        return $this->clientUserID;
    }

    public function getMeasuredUntilCleared()
    {
        return $this->measuredUntilCleared;
    }

    public function getMeasuredUntilArrived()
    {
        return $this->measuredUntilArrived;
    }

    public function getAcceptedUntilCleared()
    {
        return $this->acceptedUntilCleared;
    }

    public function getAcceptedUntilArrived()
    {
        return $this->acceptedUntilArrived;
    }

}